zokrates_embed/src/ark.rs

```diff
@@ -0,0 +1,344 @@
+use ark_bls12_377::{
+    constraints::PairingVar as BLS12PairingVar, Bls12_377 as BLS12PairingEngine, Fq as BLS12Fq,
+    Fq2 as BLS12Fq2,
+};
+use ark_bw6_761::Fr as BW6Fr;
+use ark_ec::PairingEngine;
+use ark_ff::{BigInteger, One, PrimeField};
+use ark_r1cs_std::bits::boolean::Boolean;
+use ark_relations::{
+    ns,
+    r1cs::{ConstraintSystem, ConstraintSystemRef},
+};
+
+use ark_crypto_primitives::snark::constraints::SNARKGadget;
+use ark_gm17::{constraints::GM17VerifierGadget, Proof, VerifyingKey, GM17};
+use ark_r1cs_std::alloc::{AllocVar, AllocationMode};
+
+use crate::Constraint;
+use ark_crypto_primitives::SNARK;
+use ark_r1cs_std::fields::fp::FpVar;
+use ark_r1cs_std::ToBitsGadget;
+use ark_relations::r1cs::{ConstraintSynthesizer, SynthesisError};
+use ark_std::test_rng;
+use std::str::FromStr;
+use zokrates_field::Field;
+
+type GM17Snark = GM17<BLS12PairingEngine>;
+type VerifierGadget = GM17VerifierGadget<BLS12PairingEngine, BLS12PairingVar>;
+
+type G1 = <ark_ec::bls12::Bls12<ark_bls12_377::Parameters> as PairingEngine>::G1Affine;
+type G2 = <ark_ec::bls12::Bls12<ark_bls12_377::Parameters> as PairingEngine>::G2Affine;
+
+#[derive(Copy, Clone)]
+struct DefaultCircuit {
+    pub public_input_size: usize,
+}
+
+impl<F: PrimeField> ConstraintSynthesizer<F> for DefaultCircuit {
+    fn generate_constraints(self, cs: ConstraintSystemRef<F>) -> Result<(), SynthesisError> {
+        for _ in 0..self.public_input_size {
+            let _ = FpVar::<F>::new_input(ns!(cs, "alloc_input"), || Ok(F::one()))?;
+        }
+        Ok(())
+    }
+}
+
+#[allow(clippy::type_complexity)]
+pub fn generate_verify_constraints(
+    public_input_size: usize,
+) -> (
+    usize,
+    Vec<usize>,
+    Vec<usize>,
+    Vec<usize>,
+    Vec<Constraint<BW6Fr>>,
+    usize,
+) {
+    let cs_sys = ConstraintSystem::<BW6Fr>::new();
+    let cs = ConstraintSystemRef::new(cs_sys);
+
+    let mut rng = test_rng();
```

Comment:
Does this make this function non-deterministic? If so, is it required? If not we should have a comment

Reply:
`test_rng` has a fixed seed

Reply:
Alright, can we add that as a comment? And/or change the variable name to `deterministic_rng`?

Reply:
left a comment